pkg/kontext/bundle.go

```diff
@@ -47,19 +47,49 @@ func bundle(directory string) (v1.Layer, error) {
 	buf := bytes.NewBuffer(nil)
 	tw := tar.NewWriter(buf)
 	defer tw.Close()
-
+	var excludedDirs []string = make([]string, 0)
 	err := filepath.Walk(directory,
 		func(path string, fi os.FileInfo, err error) error {
 			if err != nil {
 				return err
 			}
 
 			// Skip anything in the .git directory
-			// TODO(mattmoor): expand this to .gitignore / .dockerignore?
+
 			if fi.IsDir() && filepath.Base(path) == ".git" {
 				return filepath.SkipDir
 			}
 
+			//Check if the directory has  .dockerignore
+			hasDockerIgnore, fDockerIgnore, err := hasDockerIgnore(directory)
+
+			if err != nil {
+				return err
+			}
+
+			var patterns []string = make([]string, 0)
```

Comment:
When allocating a zero length array you can use these, which are less verbose:
```go
			// This is nil, but append(nil, foo) and len() works fine.
			var patterns []string

			// This is non-nil, but empty (good code shouldn't care about the diff)
			patterns := []string{}
```

Reply:
refactor has removed this